src/amplitude.js

```diff
@@ -137,146 +112,238 @@ Amplitude.prototype.init = function(apiKey, opt_userId, opt_config, callback) {
+ */
+Amplitude.prototype.isNewSession = function isNewSession() {
   return this._newSession;
 };
 
-Amplitude.prototype.getSessionId = function() {
+/**
+ * Returns the id of the current session.
+ * @public
+ * @return {number} Id of the current session.
+ */
+Amplitude.prototype.getSessionId = function getSessionId() {
   return this._sessionId;
 };
 
-Amplitude.prototype.nextEventId = function() {
+/**
+ * Increments the eventId and returns it.
+ * @private
+ */
+Amplitude.prototype.nextEventId = function nextEventId() {
   this._eventId++;
   return this._eventId;
 };
 
-Amplitude.prototype.nextIdentifyId = function() {
+/**
+ * Increments the identifyId and returns it.
+ * @private
+ */
+Amplitude.prototype.nextIdentifyId = function nextIdentifyId() {
   this._identifyId++;
   return this._identifyId;
 };
 
-Amplitude.prototype.nextSequenceNumber = function() {
+/**
+ * Increments the sequenceNumber and returns it.
+ * @private
+ */
+Amplitude.prototype.nextSequenceNumber = function nextSequenceNumber() {
   this._sequenceNumber++;
   return this._sequenceNumber;
 };
 
-// returns the number of unsent events and identifys
-Amplitude.prototype._unsentCount = function() {
+/**
+ * Returns the total count of unsent events and identifys
+ * @private
+ */
+Amplitude.prototype._unsentCount = function _unsentCount() {
   return this._unsentEvents.length + this._unsentIdentifys.length;
 };
 
-// returns true if sendEvents called immediately
-Amplitude.prototype._sendEventsIfReady = function(callback) {
+/**
+ * Send events if ready. Returns true if events are sent.
+ * @private
+ */
+Amplitude.prototype._sendEventsIfReady = function _sendEventsIfReady(callback) {
   if (this._unsentCount() === 0) {
     return false;
   }
 
+  // if batching disabled, send any unsent events immediately
   if (!this.options.batchEvents) {
     this.sendEvents(callback);
     return true;
   }
 
+  // if batching enabled, check if min threshold met for batch size
   if (this._unsentCount() >= this.options.eventUploadThreshold) {
     this.sendEvents(callback);
     return true;
   }
 
-  if (!this._updateScheduled) {
+  // otherwise schedule an upload after 30s
+  if (!this._updateScheduled) {  // make sure we only schedule 1 upload
     this._updateScheduled = true;
-    setTimeout(
-      function() {
+    setTimeout(function() {
         this._updateScheduled = false;
         this.sendEvents();
       }.bind(this), this.options.eventUploadPeriodMillis
     );
   }
 
-  return false;
+  return false; // an upload was scheduled, no events were uploaded
 };
 
-// storage argument allows for localStorage and sessionStorage
-Amplitude.prototype._getFromStorage = function(storage, key) {
+/**
+ * Helper function to fetch values from storage
+ * Storage argument allows for localStoraoge and sessionStoraoge
+ * @private
+ * @deprecated
+ */
+Amplitude.prototype._getFromStorage = function _getFromStorage(storage, key) {
```

Comment:
Scratch the above comment, I forgot it's used in multiple apps. The important piece is missing from the code, namely the part where we append the namespace identifier to the key